src/openforms/submissions/tasks/__init__.py

```diff
@@ -22,127 +21,50 @@
 logger = logging.getLogger(__name__)
 
 
-def on_completion(submission_id: int) -> None:
-    """
-    Celery chain of tasks to execute on a submission completion event.
+def on_post_submission_event(submission_id: int) -> None:
+    # this can run any time because they have been claimed earlier
```

Comment:
Can you restore the docstring as much as possible?

The part about `transaction.on_commit` is quite relevant still.